crates/turbopack/src/module_options/rule_condition.rs

```diff
@@ -20,6 +20,18 @@ pub enum ModuleRuleCondition {
     ResourcePathInDirectory(String),
     ResourcePathInExactDirectory(FileSystemPathReadRef),
     ResourcePathRegex(#[turbo_tasks(trace_ignore)] Regex),
+    /// For paths that are within the same filesystem as the `base`, it need to
+    /// match the relative path from base to resource. This includes `./` or
+    /// `../` prefix. For paths in a different filesystem, it need to match
+    /// the resource path in that filesystem without any prefix. This means
+    /// any glob starting with `./` or `../` will only match paths in the
+    /// project. Globs starting with `**` can match any path.
```

Comment:
Because we're forcing the path to be relative with a leading `./` or `../` with a `get_relative_path_to` call, I think this breaks a lot of cases? Like, I don't write my globs with `./foo/**/*.js`, I just write `foo/**/*.js`.

Reply:
Yeah, I was also thinking this might be a source of errors.

Do you have a better idea for that?

These are the use cases I had in mind:

```
A: *.png - matching any file with that extension
B: ./*.png - matching png files in the project directory
C: ../other-package/**/*.png - matching png files in a different package
D: ./sub/dir/*.png - matching png files in a sub directory
E: **/dir/*.png - matching png files that are in "dir" anywhere
```

If we omit the prefix A and B are conflicting.

Reply:
I think you're cases are correct, except for `D`. I would have written that `sub/dir/*.png`. We could make the globs relative if they contain a `/` and don't begin with `.`?

Reply:
yes possible. Might be a bit unexpected in this edge case `{./,dir}/sub/*.png`, but I can live with that